### ISSUE TITLE
No cognito user attributes in authData object 

### ISSUE THREAD

**Issue by kodeinEU** [aws-amplify/amplify-js#1728]
Hi, I'm using standard amplify/cognito authentication flow in my react app. 
After SignIn with SMS confirmation my App render function needs to do some checks on signin user attributes. The App component is wrapped with withAuthenticator, so I can read authData from my props.
But there is no attributes inside authData object. After page refresh (F5) attributes are there as expected.

the authData object in standard authentication flow (SignIn->ConfirmSignIn->App)  looks like this:
```
Session: "..."
authenticationFlowType: "USER_SRP_AUTH"
challengeName: "SMS_MFA"
challengeParam: {CODE_DELIVERY_DELIVERY_MEDIUM: "SMS", CODE_DELIVERY_DESTINATION: "..."}
client: Client {endpoint: "...", userAgent: "aws-amplify/0.1.x js"}
pool: CognitoUserPool {userPoolId: "...", clientId: "...", client: Client, advancedSecurityDataCollectionFlag: true, storage: Storage}
signInUserSession: CognitoUserSession {idToken: CognitoIdToken, refreshToken: CognitoRefreshToken, accessToken: CognitoAccessToken, clockDrift: 0}
storage: ...
username: "..."
```

Now when I reload page (F5) signedin user attributes are added to authData object and it looks like this
```
Session: null
attributes: {email_verified: false, name: "John", …}
authenticationFlowType: "USER_SRP_AUTH"
client: Client {endpoint: "...", userAgent: "..."}
pool: CognitoUserPool {userPoolId: "...", clientId: "...", client: Client, advancedSecurityDataCollectionFlag: true, storage: Storage}
preferredMFA: "NOMFA"
signInUserSession: CognitoUserSession {idToken: CognitoIdToken, refreshToken: CognitoRefreshToken, accessToken: CognitoAccessToken, clockDrift: 0}
storage: ...
username: ""
```

Any idea why this happens??

thanks, 
Wojtek

**Comment by powerful23** [comment]
@kodeinEU that's caused because when the page is reloaded, the `Auth.currentAuthenticatedUser` is called and returned the attributes to you.

**Comment by vdotniemykin** [comment]
Are there any ways to display user's email or name just after signing in?
Thank you in advance.

**Comment by powerful23** [comment]
Closing the issue due to no further response. Fell free to reopen if you still have this issue.

**Comment by ron23** [comment]
@powerful23 any suggestions to OP's latest question?
> Are there any ways to display user's email or name just after signing in?

Thanks!

**Comment by kicoo7** [comment]
Still no answer for this???? :hocho:

**Comment by adrianmaurer** [comment]
I'm having a similar issue. Wish the attributes would be available when the user object is updated and at login

**Comment by powerful23** [comment]
@adrianmaurer this issue should have been fixed in #2918 , are you still facing this issue with the latest version?

**Comment by CMMR2** [comment]
I came across this issue after experiencing the problem on a new project I started last week. I'm using the latest version and the attributes are definitely not there until a page refresh.
